include/unifex/let_value_with_stop_token.hpp

```diff
@@ -0,0 +1,234 @@
+                                  Receiver&&>)
+      -> operation<
+          member_t<Self, SuccessorFactory>,
+          unifex::remove_cvref_t<Receiver>> {
+    return operation<
+        member_t<Self, SuccessorFactory>,
+        unifex::remove_cvref_t<Receiver>>(
+        static_cast<Self&&>(self).func_, static_cast<Receiver&&>(r));
+  }
+
+private:
+  UNIFEX_NO_UNIQUE_ADDRESS SuccessorFactory func_;
+};
+
+template <typename SuccessorFactory, typename Receiver>
+struct _stop_token_operation<SuccessorFactory, Receiver>::type {
+  using inner_sender_t =
+      unifex::invoke_result_t<SuccessorFactory&&, inplace_stop_token&>;
+  using receiver_t = stop_token_receiver<
+      operation<SuccessorFactory, unifex::remove_cvref_t<Receiver>>,
+      unifex::remove_cvref_t<Receiver>>;
+
+  static constexpr bool successor_is_nothrow =
+      unifex::is_nothrow_invocable_v<SuccessorFactory&&, inplace_stop_token&>;
+  template <typename Receiver2>
+  static constexpr bool inner_receiver_nothrow_constructible = unifex::
+      is_nothrow_constructible_v<receiver_t, inplace_stop_token, Receiver2&&>;
+  static constexpr bool nothrow_connectable =
+      unifex::is_nothrow_connectable_v<inner_sender_t, receiver_t>;
+
+private:
+  template <typename Receiver2>
+  auto connect_inner_op(
+      SuccessorFactory&& func,
+      inplace_stop_token st,
+      Receiver2&&
+          r) noexcept(successor_is_nothrow&&
+                          inner_receiver_nothrow_constructible<Receiver2>&&
+                              unifex::is_nothrow_connectable_v<
+                                  inner_sender_t,
+                                  receiver_t>) {
+    if constexpr (
+        successor_is_nothrow &&
+        inner_receiver_nothrow_constructible<Receiver2> &&
+        unifex::is_nothrow_connectable_v<inner_sender_t, receiver_t>) {
+      return unifex::connect(
+          ((SuccessorFactory &&) func)(st), receiver_t(st, (Receiver2 &&) r));
+    } else {
+      // Need to manually unsubscribe stop_token_adapter_ in case call to
+      // connect throws
+      UNIFEX_TRY {
+        return unifex::connect(
+            ((SuccessorFactory &&) func)(st), receiver_t(st, (Receiver2 &&) r));
+      }
+      UNIFEX_CATCH(...) {
+        stop_token_adapter_.unsubscribe();
+        throw;
+      }
+    }
+  }
+
+  inplace_stop_token_adapter<stop_token_type_t<Receiver>> stop_token_adapter_;
+  connect_result_t<inner_sender_t, receiver_t> innerOp_;
+
+public:
+  template <typename SuccessorFactory2, typename Receiver2>
+  type(SuccessorFactory2&& func, Receiver2&& r) noexcept(
+      is_nothrow_constructible_v<SuccessorFactory, SuccessorFactory2&&>&&
+          is_nothrow_constructible_v<std::decay_t<Receiver2>, Receiver2>&& noexcept(
+              connect_inner_op(
+                  std::declval<SuccessorFactory2&&>(),
+                  std::declval<inplace_stop_token>(),
+                  std::declval<Receiver2&&>())))
+    : innerOp_(connect_inner_op(
```

Comment:
GCC seems to want some `this`es.  Also, since the `noexcept` clause follows the argument list, you can refer to the arguments here instead of using `declval`, which I find clearer.

```suggestion
              this->connect_inner_op(
                  (SuccessorFactory &&) func,
                  stop_token_adapter_.subscribe(get_stop_token(r)),
                  (Receiver2 &&) r)))
    : innerOp_(connect_inner_op(
```